addons/mail/wizard/mail_compose_message.py

```diff
@@ -203,6 +203,40 @@ def _check_res_domain(self):
         for composer in self:
             composer._evaluate_res_domain()
 
+    @api.depends('composition_mode', 'model', 'parent_id', 'res_ids',
+                 'template_id')
+    def _compute_subject(self):
+        """ Computation is coming either form template, either from context.
+        When having a template with a value set, copy it (in batch mode) or
+        render it (in monorecord comment mode) on the composer. Otherwise
+        it comes from the parent (if set), or computed based on the generic
+        '_message_compute_subject' method in monorecord comment mode, or
+        set to False. When removing the template, reset it. """
+        for composer in self:
+            if composer.template_id:
+                composer._set_value_from_template('subject', force_void=False)
+            if (not composer.template_id or not composer.subject) and composer.composition_mode == 'mass_mail':
+                composer.subject = False
+            elif not composer.template_id or not composer.subject:
+                subject = composer.parent_id.subject
+                if not subject:
+                    res_ids = composer._evaluate_res_ids()
+                    if len(res_ids) == 1:
+                        subject = self.env[composer.model].browse(res_ids)._message_compute_subject()
+                composer.subject = subject
+
+    @api.depends('composition_mode', 'model', 'res_ids', 'template_id')
+    def _compute_body(self):
+        """ Computation is coming either from template, either reset. When
+        having a template with a value set, copy it (in batch mode) or render
+        it (in monorecord comment mode) on the composer. When removing the
+        template, reset it. """
+        for composer in self:
+            if composer.template_id and not is_html_empty(composer.template_id.body_html):
+                composer._set_value_from_template('body_html', 'body', force_void=False)
```

Comment:
Should "_set_value_from_template" use "is_html_empty" itself?

Would make sense otherwise the void option is a bit useless, as it can set empty content.